packages/core/src/2d/text/TextUtils.ts

```diff
@@ -110,52 +110,135 @@ export class TextUtils {
       let charsWidth = 0;
       let maxAscent = -1;
       let maxDescent = -1;
+      let wordChars = "";
+      let wordCharsWidth = 0;
+      let wordMaxAscent = -1;
+      let wordMaxDescent = -1;
+      let curInEnglishWord = false;
 
       for (let j = 0, m = subText.length; j < m; ++j) {
         const char = subText[j];
         const charInfo = TextUtils._getCharInfo(char, fontString, subFont);
+        const charCode = char.charCodeAt(0);
+        // 32 is space.
+        const isNotEnglish = charCode === 32 || charCode > 255;
         const { w, offsetY } = charInfo;
         const halfH = charInfo.h * 0.5;
         const ascent = halfH + offsetY;
         const descent = halfH - offsetY;
-        if (charsWidth + w > wrapWidth) {
-          if (charsWidth === 0) {
-            lines.push(char);
-            lineWidths.push(w);
-            lineMaxSizes.push({
-              ascent,
-              descent,
-              size: ascent + descent
-            });
+
+        if (isNotEnglish) {
+          // If it is an English word before, need to handle the previous English word and chars.
+          if (curInEnglishWord) {
+            if (charsWidth + wordCharsWidth > wrapWidth) {
```

Comment:
I think the definition of `wrapWidth` should not be `renderer.width`, he should use a new variable definition, like `wordWrapWidth`.

Reply:
I think this will make things more complex, but the local bounds should discuss.